test/system/SmokeTestHostless.cpp

```diff
@@ -153,18 +155,21 @@ BOOST_AUTO_TEST_CASE(hostless_server) {
             // if first game lobby should be empty
             BOOST_REQUIRE_EQUAL(GetLobbyAICount(), 0);
 
+            BOOST_TEST_MESSAGE("Filling lobby with " << num_AIs << "AIs.");
             // fill lobby with AIs
             for (unsigned int ai_i = 1; ai_i <= num_AIs; ++ai_i) {
                 PlayerSetupData ai_plr;
                 ai_plr.client_type = Networking::ClientType::CLIENT_TYPE_AI_PLAYER;
-                m_lobby_data.players.push_back({Networking::INVALID_PLAYER_ID, ai_plr});
+                m_lobby_data.players.emplace_back(Networking::INVALID_PLAYER_ID, ai_plr);
```

Comment:
Could it cause issues with client type?

Reply:
If you mean could it cause the error I see, no, I don't think so. These lines should be equivalent except the `emplace` version might be slightly more efficient by constructing in place rather than constructing then moving. The client type issue also probably is there before these changes, as I was adding this logging as part of trying to figure out why the test fails.